aws_lambda_builders/workflows/python_pip/workflow.py

```diff
@@ -54,17 +59,25 @@ def __init__(self,
                  artifacts_dir,
                  scratch_dir,
                  manifest_path,
-                 runtime=None, **kwargs):
+                 runtime_path=None,
+                 runtime=None,
+                 **kwargs):
+
+        self.runtime_path = PathResolver(language=self.CAPABILITY.language, runtime=runtime).path
 
         super(PythonPipWorkflow, self).__init__(source_dir,
                                                 artifacts_dir,
                                                 scratch_dir,
                                                 manifest_path,
+                                                runtime_path=self.runtime_path,
```

Comment:
I understand what you are doing here, but the classes seem to be coupled tightly thru the constructor. Through the constructor you get inputs that are common to every workflow. These inputs are agnostic of how the specific workflow works. 

But items like `runtime_path` are highly dependent on the workflow's implementation. In fact, i can imagine a scenario where a workflow conditionally chooses different `runtime_path` values when the workflow is actually being run. 

Can you explore a different design pattern instead?

In my mind, the Workflows to follow a [Plugin design pattern](https://martinfowler.com/eaaCatalog/plugin.html)

Reply:
Yeah one of the other actions we can explore is runtime validation being done as part of an action, and pass state via callbacks. The state in this case, is the path is still required for another build action. This could be the same with another language too.